csharp/Microsoft.Azure.Databricks.Client/Models/InitScriptInfo.cs

```diff
@@ -12,4 +12,10 @@ public record InitScriptInfo
     /// </summary>
     [JsonPropertyName("dbfs")]
     public DbfsStorageInfo Dbfs { get; set; }
+
+    /// <summary>
+    /// Workspace location of init script. destination must be provided.
+    /// </summary>
+    [JsonPropertyName("workspace")]
+    public WorkspaceStorageInfo Workspace { get; set; }
```

Comment:
The databricks rest API doc wasn't very clear on this, but the InitScriptInfo object should contain **one of** "dbfs", "workspace" or "volumns" properties. List both Dbfs and Workspace as property here won't enforce the "one of" design. I'm thinking of a similar design as the [Library classes](https://github.com/Azure/azure-databricks-client/blob/master/csharp/Microsoft.Azure.Databricks.Client/Models/Library.cs), with a [custom converter](https://github.com/Azure/azure-databricks-client/blob/master/csharp/Microsoft.Azure.Databricks.Client/Converters/LibraryConverter.cs).

Reply:
Where did you find the information about init scripts being implemented as "one of" design? I saw jobs configured in our environment where there was a DBFS init script as well as a workspace init script defined when i was trying to migrate since DBFS init scripts are deprecated. It would not make sense to design these as "one of" here, IMO

Reply:
I inferred from the schema. "One of" here refers to individual object in the "init_scripts" array - within each object, you can specify "one of" workspace, volumes and dbfs (you can still have multiple objects in the array).
```json
"init_scripts": [
  {
    "workspace": {...}
  },
  {
    "workspace": {...}
  },
  {
    "volumes": {...}
  },
  {
    "dbfs": {...}
  },
  {
    "dbfs": {...}
  }
]
```
vs
```json
// I believe this should be invalid
"init_scripts": [
  {
    "workspace": {...}
    "volumes": {...}
    "dbfs": {...}
  }
]
```
Let me apply your PR and test it out first.

Reply:
Okay I tested your changes. Here is a counter example:
```cs
var clusterConfig = ClusterAttributes
    .GetNewClusterConfiguration("SampleProgram cluster")
    .WithRuntimeVersion(RuntimeVersions.Runtime_10_4)
    .WithAutoTermination(30)
    .WithNodeType(NodeTypes.Standard_D3_v2)
    .WithClusterMode(ClusterMode.SingleNode);
clusterConfig.InitScripts = new List<InitScriptInfo> {
    new InitScriptInfo {
        Dbfs = new InitScriptStorageInfo { Destination = "dbfs:/test" },
        Workspace = new InitScriptStorageInfo {Destination = "/Users/user1@databricks.com/my-init.sh"}
    }
};

var clusterId = await client.Clusters.Create(clusterConfig);
```
If you inspect the created cluster, you'll notice that only the "workspace" type of init_script is created, and the "dbfs" type is ignored. This is because the fields in the InitScriptInfo class is supposed to be "one of". The API design does not prevent user from specifying more than one fields, but only one field will be respected by the REST API.

Reply:
@jonathan-vogel-siemens let me know if you still want to give it a try and use a similar design as the [Library classes](https://github.com/Azure/azure-databricks-client/blob/master/csharp/Microsoft.Azure.Databricks.Client/Models/Library.cs), with a [custom converter](https://github.com/Azure/azure-databricks-client/blob/master/csharp/Microsoft.Azure.Databricks.Client/Converters/LibraryConverter.cs). Otherwise I can create a new PR and take care of the changes.

Reply:
You are right, thanks for trying it out. I think it would make sense to implement the pattern as seen in the Library classes then.

Reply:
@jonathan-vogel-siemens are you still going to implement the requested changes to this PR? If not, I'm happy to take over the work and continue with the changes. I hope to wrap up this PR and release a new version (v2.2.0) soon.